pkg/standalone-utils/contracts/interfaces/IBaseRelayerLibrary.sol

```diff
@@ -23,6 +23,20 @@ import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
 abstract contract IBaseRelayerLibrary {
     function getVault() public view virtual returns (IVault);
 
+    function approveVault(IERC20 token, uint256 amount) public virtual;
```

Comment:
I didn't see the need for this before, but now I do: if the relayer might be sender, it needs to have allowance.

The problem with this interface is that we'd always provide infinite allowance, as we wouldn't be able to pass a chained reference here. Would it make sense to instead have swap, join, etc. all do manual approvals for the exact amount before that operation if the relayer was the sender? It'd be extra work for us (meaning we'll need to add functionality to those functions), but way more gas efficient and easier to use for the front-end.

Reply:
Thoughts on adding a `_peekChainedReferenceValue()` function for cases like this? This would read a referenced value but not zero it out.

Reply:
The issue with bundling an approval into the vault actions is that it then requires that we disallow any GIVEN_OUT actions as these would leave dust approvals on the vault. (unless we purposefully zeroed out the allowance afterwards.)

Reply:
Let's keep infinite allowance for now. Created https://github.com/balancer-labs/balancer-v2-monorepo/issues/991